study1/week3/reduce-array-size-to-the-half.java

```diff
@@ -0,0 +1,44 @@
+//https://leetcode.com/problems/reduce-array-size-to-the-half/description/
+
+class Solution {
+    public int minSetSize(int[] arr) {
+        //time complexity is O(n)
+        //space complexity is also O(n)
+        
+        int arrLength = arr.length;
```

Comment:
Please move these two variable on [where it's used](https://google.github.io/styleguide/javaguide.html#s4.8.2-variable-declarations) as mentioned in the style guide, so in this case we should move it to line 31.